CMakeLists.txt

```diff
@@ -91,18 +95,44 @@ endif()
 ###
 
 # External libraries
-find_package(OpenSSL REQUIRED)
+
+if ( BORINGSSL_ROOT_DIR )
+   find_package(BoringSSL REQUIRED)
+
+   if ( BORINGSSL_FOUND )
+      add_compile_definitions(WITH_BORINGSSL)
+      add_compile_options(-Wno-gnu-anonymous-struct -Wno-nested-anon-types)
+
+      message(STATUS "BoringSSL Found: ${BORINGSSL_COMPAT_VERSION}")
+      message(STATUS "BoringSSL Include: ${BORINGSSL_INCLUDE_DIR}")
+      message(STATUS "BoringSSL Libraries: ${BORINGSSL_SSL_LIBRARY};${BORINGSSL_CRYPTO_LIBRARY}")
+   else ()
+      message(FATAL_ERROR "BORINGSSL_ROOT_DIR set but no BoringSSL library found")
+   endif ()
+
+   set(OPENSSL_ROOT_DIR ${BORINGSSL_ROOT_DIR})
+   set(OPENSSL_INCLUDE_DIR ${BORINGSSL_INCLUDE_DIR})
+   set(OPENSSL_CRYPTO_LIBRARY ${BORINGSSL_CRYPTO_LIBRARY})
+   set(OPENSSL_SSL_LIBRARY ${BORINGSSL_SSL_LIBRARY})
+   
+   find_package(OpenSSL REQUIRED)
+
+   set(OPENSSL_VERSION ${BORINGSSL_COMPAT_VERSION})
```

Comment:
We need to override this after calling `find_package(OpenSSL REQUIRED)` since it gets blanked out otherwise.